internal/controller/controllers/local_cluster_import_controller.go

```diff
@@ -0,0 +1,491 @@
+}
+
+func (r *LocalClusterImportReconciler) DeleteClusterDeployment(ctx context.Context, namespace string, name string) error {
+	clusterDeployment := &hivev1.ClusterDeployment{}
+	namespacedName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}
+	err := r.Client.Get(ctx, namespacedName, clusterDeployment)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrapf(err, "error fetching ClusterDeployment %s in namespace %s for delete", name, namespace)
+	}
+	err = r.Client.Delete(ctx, clusterDeployment)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrapf(err, "error deleting ClusterDeployment %s in namespace %s", name, namespace)
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) DeleteAgentClusterInstall(ctx context.Context, namespace string, name string) error {
+	agentClusterInstall := &hiveext.AgentClusterInstall{}
+	namespacedName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}
+	err := r.Client.Get(ctx, namespacedName, agentClusterInstall)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrapf(err, "error fetching AgentClusterInstall %s in namespace %s for delete", name, namespace)
+	}
+	err = r.Client.Delete(ctx, agentClusterInstall)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Wrapf(err, "error deleting AgentClusterInstall %s in namespace %s", name, namespace)
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) HasLocalManagedCluster(ctx context.Context) (bool, error) {
+	managedCluster := &clusterv1.ManagedCluster{}
+	namespacedName := types.NamespacedName{
+		Namespace: "",
+		Name:      r.LocalClusterName,
+	}
+	err := r.Client.Get(ctx, namespacedName, managedCluster)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func (r *LocalClusterImportReconciler) createAgentClusterInstall(ctx context.Context, numberOfControlPlaneNodes int, proxy *configv1.Proxy) error {
+	//Create an AgentClusterInstall in the local cluster namespace
+	userManagedNetworkingActive := true
+	agentClusterInstall := &hiveext.AgentClusterInstall{
+		Spec: hiveext.AgentClusterInstallSpec{
+			Networking: hiveext.Networking{
+				UserManagedNetworking: &userManagedNetworkingActive,
+			},
+			ClusterDeploymentRef: v1.LocalObjectReference{
+				Name: r.LocalClusterName,
+			},
+			ImageSetRef: &hivev1.ClusterImageSetReference{
+				Name: "local-cluster-image-set",
+			},
+			ProvisionRequirements: hiveext.ProvisionRequirements{
+				ControlPlaneAgents: numberOfControlPlaneNodes,
+			},
+		},
+	}
+	agentClusterInstall.Namespace = r.LocalClusterName
+	agentClusterInstall.Name = r.LocalClusterName
+	if proxy != nil {
+		agentClusterInstall.Spec.Proxy = &hiveext.Proxy{
+			HTTPProxy:  proxy.Spec.HTTPProxy,
+			HTTPSProxy: proxy.Spec.HTTPSProxy,
+			NoProxy:    proxy.Spec.NoProxy,
+		}
+	}
+
+	err := r.Client.Create(ctx, agentClusterInstall)
```

Comment:
I think this should probably be using `controllerutil.CreateOrUpdate` to ensure we keep our desired state.